src/Microsoft.ML.Tokenizers/Model/CodeGen.cs

```diff
@@ -0,0 +1,1723 @@
+        {
+            { "<|endoftext|>",                      50256 },
+            { "                               ",    50257 },
+            { "                              ",     50258 },
+            { "                             ",      50259 },
+            { "                            ",       50260 },
+            { "                           ",        50261 },
+            { "                          ",         50262 },
+            { "                         ",          50263 },
+            { "                        ",           50264 },
+            { "                       ",            50265 },
+            { "                      ",             50266 },
+            { "                     ",              50267 },
+            { "                    ",               50268 },
+            { "                   ",                50269 },
+            { "                  ",                 50270 },
+            { "                 ",                  50271 },
+            { "                ",                   50272 },
+            { "               ",                    50273 },
+            { "              ",                     50274 },
+            { "             ",                      50275 },
+            { "            ",                       50276 },
+            { "           ",                        50277 },
+            { "          ",                         50278 },
+            { "         ",                          50279 },
+            { "        ",                           50280 },
+            { "       ",                            50281 },
+            { "      ",                             50282 },
+            { "     ",                              50283 },
+            { "    ",                               50284 },
+            { "   ",                                50285 },
+            { "  ",                                 50286 },
+            { "\t\t\t\t\t\t\t\t\t",                 50287 },
+            { "\t\t\t\t\t\t\t\t",                   50288 },
+            { "\t\t\t\t\t\t\t",                     50289 },
+            { "\t\t\t\t\t\t",                       50290 },
+            { "\t\t\t\t\t",                         50291 },
+            { "\t\t\t\t",                           50292 },
+            { "\t\t\t",                             50293 },
+            { "\t\t",                               50294 },
+        };
+
+        private static Dictionary<StringSpanOrdinalKey, (int, string)> GetVocabulary(Stream vocabularyStream)
+        {
+            Dictionary<StringSpanOrdinalKey, (int, string)>? vocab;
+            try
+            {
+                JsonSerializerOptions options = new() { Converters = { StringSpanOrdinalKeyCustomConverter.Instance } };
+                vocab = JsonSerializer.Deserialize<Dictionary<StringSpanOrdinalKey, (int, string)>>(vocabularyStream, options) as Dictionary<StringSpanOrdinalKey, (int, string)>;
```

Comment:
I assume the expectation is this code is executing rarely / basically once per process?

Reply:
This code is executed once for each instantiation of the CodeGen Tokenizer. It reads the vocabulary file utilized by the tokenizer.